src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.QueueItem.cs

```diff
@@ -5,59 +5,95 @@
 #nullable enable
 
 using System;
-using System.Collections.Immutable;
 using System.Diagnostics;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Threading;
 
 namespace Microsoft.CodeAnalysis.LanguageServer.Handler
 {
     internal partial class RequestExecutionQueue
     {
-        private readonly struct QueueItem
+        private interface QueueItem
         {
             /// <summary>
-            /// Callback to call into underlying <see cref="IRequestHandler"/> to perform the actual work of this item.
+            /// Begins executing the work specified by this queue item.
             /// </summary>
-            private readonly Func<RequestContext?, CancellationToken, Task> _callbackAsync;
+            public Task CallbackAsync(RequestContext? context, CancellationToken cancellationToken);
 
             /// <summary>
-            /// <see cref="CorrelationManager.ActivityId"/> used to properly correlate this work with the loghub
-            /// tracing/logging subsystem.
+            /// Sets the result of this queue item to an exception.  Used if we fail to enqueue this item before it starts.
             /// </summary>
-            public readonly Guid ActivityId;
-            private readonly ILspLogger _logger;
+            public bool TrySetException(Exception e);
```

Comment:
on the fence if we should just expose teh TCS.

Reply:
I don't think I can expose the tcs since the tcs is generic, but in the queue, after de-queueing it isn't generic and unknown what to cast it to

Reply:
Great point!